src/game/state.ts

```diff
@@ -33,6 +33,15 @@ export const checkIfGameWon = (gameState: GameState, board: Board): boolean => {
   return false;
 };
 
+export const updateAllowedTiles = (board: Board): Board => {
+  const newBoard = JSON.parse(JSON.stringify(board));
```

Comment:
Why not using spraed operator here ?

```suggestion
  const newBoard = [...board];
```

Reply:
Spread operator only makes shallow clones of objects.
But since function is removed anyway, I mark this as resolved.